ftw/calendarexport/export.py

```diff
@@ -24,17 +25,29 @@ class ExportICS(CalendarView):
     def update(self):
         context = aq_inner(self.context)
         catalog = getToolByName(context, 'portal_catalog')
-        provides = ICalendarSupport.__identifier__
         uids = self.request.form.get('uids', [])
-        self.events = catalog(UID=uids, object_provides=provides)
+        self.events = catalog(UID=uids)
         if not uids:
             self.events = []
 
     @ram.cache(cachekey)
     def feeddata(self):
         data = cs.ICS_HEADER % dict(prodid=cs.PRODID)
         for brain in self.events:
-            tmp_data = brain.getObject().getICal()
-            data += tmp_data
+            obj = brain.getObject()
+            event_data = ""
+
+            if ICalendarSupport.providedBy(obj):
+                event_data = obj.getICal()
+
+            if PLONE_APP_EVENTS_AVAILABLE:
+                from plone.app.event.dx.interfaces import IDXEvent
+                from plone.app.event.ical import construct_icalendar
+                if IDXEvent.providedBy(obj):
+                    cal = construct_icalendar(obj, obj)
+                    event = cal.subcomponents[0]
+                    event_data = event.to_ical()
+
+            data += event_data
```

Comment:
I propose to change `data` to a list for [performance optimization reasons](https://wiki.python.org/moin/PythonSpeed/PerformanceTips#String_Concatenation):
```python
data = []
for ..:
  data.append(event_data)
return ''.join(data)
```